src/sage/repl/preparse.py

```diff
@@ -1189,6 +1190,15 @@ def preparse_numeric_literals(code, extract=False, quotes="'"):
         sage: preparse_numeric_literals('000042')
         'Integer(42)'
 
+    Check that :issue:`40179` is fixed::
+
+        sage: preparse_numeric_literals("1" * 4300) == f"Integer({"1" * 4300})"
```

Comment:
```suggestion
        sage: preparse_numeric_literals("1" * 4300) == f"Integer({'1' * 4300})"
```
This test is broken because the inner quote on the RHS needs to be different (or escaped `\"`).

Reply:
interestingly it's new feature in 3.12 https://stackoverflow.com/a/77858818/5267751 . Lowest supported version is (still) 3.11 so yes.